pipeline/pipeline/backend.py

```diff
@@ -91,14 +95,17 @@ def copy_input(r):
                 script += task._command + ['\n']
 
         def write_pipeline_outputs(r, dest):
-            dest = os.path.abspath(dest)
-            directory = os.path.dirname(dest)
-            os.makedirs(directory, exist_ok=True)
+            if not dest.startswith('gs://'):
+                dest = os.path.abspath(dest)
+                directory = os.path.dirname(dest)
+                os.makedirs(directory, exist_ok=True)
+
+            cp = 'gsutil cp' if dest.startswith("gs://") else 'cp'
```

Comment:
I think it would make more sense to fold this into the existing `if not`

```python
if not dest.startswith('gs://'):
  # ...
   cp = 'cp'
else:
   cp = 'gsutil cp'

```